### ISSUE TITLE
Readyonly app

### ISSUE THREAD

**Issue by tuxity** [mattermost/docker#18]
Hello,

I tried to use this new docker setup (which is awesome btw, super clean) and at startup of Mattermost I had an error

```
2021-07-30T13:19:47.738Z        error   web/static.go:26        Failed to update assets subpath from config     {"error": "failed to update root.html with subpath /: open /mattermost/client/root.html: read-only file system", "errorVerbose": "open /mattermost/client/root.html: read-only file system\nfailed to update root.html with subpath /\ngithub.com/mattermost/mattermost-server/v5/utils.UpdateAssetsSubpathInDir\n\tgithub.com/mattermost/mattermost-server/v5/utils/subpath.go:117\ngithub.com/mattermost/mattermost-server/v5/utils.UpdateAssetsSubpath\n\tgithub.com/mattermost/mattermost-server/v5/utils/subpath.go:145\ngithub.com/mattermost/mattermost-server/v5/utils.UpdateAssetsSubpathFromConfig\n\tgithub.com/mattermost/mattermost-server/v5/utils/subpath.go:168\ngithub.com/mattermost/mattermost-server/v5/web.(*Web).InitStatic\n\tgithub.com/mattermost/mattermost-server/v5/web/static.go:25\ngithub.com/mattermost/mattermost-server/v5/web.New\n\tgithub.com/mattermost/mattermost-server/v5/web/web.go:36\ngithub.com/mattermost/mattermost-server/v5/cmd/mattermost/commands.runServer\n\tgithub.com/mattermost/mattermost-server/v5/cmd/mattermost/commands/server.go:101\ngithub.com/mattermost/mattermost-server/v5/cmd/mattermost/commands.serverCmdF\n\tgithub.com/mattermost/mattermost-server/v5/cmd/mattermost/commands/server.go:59\ngithub.com/spf13/cobra.(*Command).execute\n\tgithub.com/spf13/cobra@v1.1.3/command.go:852\ngithub.com/spf13/cobra.(*Command).ExecuteC\n\tgithub.com/spf13/cobra@v1.1.3/command.go:960\ngithub.com/spf13/cobra.(*Command).Execute\n\tgithub.com/spf13/cobra@v1.1.3/command.go:897\ngithub.com/mattermost/mattermost-server/v5/cmd/mattermost/commands.Run\n\tgithub.com/mattermost/mattermost-server/v5/cmd/mattermost/commands/root.go:14\nmain.main\n\tgithub.com/mattermost/mattermost-server/v5/cmd/mattermost/main.go:31\nruntime.main\n\truntime/proc.go:204\nruntime.goexit\n\truntime/asm_amd64.s:1374"}
```

I fixed it by commenting out https://github.com/mattermost/docker/blob/main/docker-compose.yml#L37 not use if it's the right fix

**Comment by wornet-aer** [comment]
I just ran into the same issue, your fix made it working again. Thanks for pointing this out!

**Comment by elandorr** [comment]
I concur! 2 weeks and no response from the mattermost team, however.

**Comment by mrckndt** [comment]
> Hello,
> 
> I tried to use this new docker setup (which is awesome btw, super clean) and at startup of Mattermost I had an error
> 
> ```
> 2021-07-30T13:19:47.738Z        error   web/static.go:26        Failed to update assets subpath from config     {"error": "failed to update root.html with subpath /: open /mattermost/client/root.html: read-only file system", "errorVerbose": "open /mattermost/client/root.html: read-only file system\nfailed to update root.html with subpath /\ngithub.com/mattermost/mattermost-server/v5/utils.UpdateAssetsSubpathInDir\n\tgithub.com/mattermost/mattermost-server/v5/utils/subpath.go:117\ngithub.com/mattermost/mattermost-server/v5/utils.UpdateAssetsSubpath\n\tgithub.com/mattermost/mattermost-server/v5/utils/subpath.go:145\ngithub.com/mattermost/mattermost-server/v5/utils.UpdateAssetsSubpathFromConfig\n\tgithub.com/mattermost/mattermost-server/v5/utils/subpath.go:168\ngithub.com/mattermost/mattermost-server/v5/web.(*Web).InitStatic\n\tgithub.com/mattermost/mattermost-server/v5/web/static.go:25\ngithub.com/mattermost/mattermost-server/v5/web.New\n\tgithub.com/mattermost/mattermost-server/v5/web/web.go:36\ngithub.com/mattermost/mattermost-server/v5/cmd/mattermost/commands.runServer\n\tgithub.com/mattermost/mattermost-server/v5/cmd/mattermost/commands/server.go:101\ngithub.com/mattermost/mattermost-server/v5/cmd/mattermost/commands.serverCmdF\n\tgithub.com/mattermost/mattermost-server/v5/cmd/mattermost/commands/server.go:59\ngithub.com/spf13/cobra.(*Command).execute\n\tgithub.com/spf13/cobra@v1.1.3/command.go:852\ngithub.com/spf13/cobra.(*Command).ExecuteC\n\tgithub.com/spf13/cobra@v1.1.3/command.go:960\ngithub.com/spf13/cobra.(*Command).Execute\n\tgithub.com/spf13/cobra@v1.1.3/command.go:897\ngithub.com/mattermost/mattermost-server/v5/cmd/mattermost/commands.Run\n\tgithub.com/mattermost/mattermost-server/v5/cmd/mattermost/commands/root.go:14\nmain.main\n\tgithub.com/mattermost/mattermost-server/v5/cmd/mattermost/main.go:31\nruntime.main\n\truntime/proc.go:204\nruntime.goexit\n\truntime/asm_amd64.s:1374"}
> ```
> 
> I fixed it by commenting out https://github.com/mattermost/docker/blob/main/docker-compose.yml#L37 not use if it's the right fix

@tuxity, @wornet-aer you're right you can circumvent the error by making the container writable again. From a security hardening standpoint i'm trying to avoid this and we're searching for a better solution. For now you can ignore this especially when you use the default setup/Nginx config and use a (sub)domain instead of a subpath other than '/'; this is failing in a function which changes this subpath for the static assets. At the moment the regex [here](https://github.com/mattermost/mattermost-server/blob/96593580aeea0bc3a4780e5284ab199f5e19bc83/utils/subpath.go#L88) removes a whitespace from the meta tag which does is harmless; it doesn't mean it shouldn't be fixed though.

**Comment by tuxity** [comment]
Ok I will comment out the read-only line for now. Thanks for the information!

**Comment by cobenash** [comment]
I didn't face the same issue.

I use a new ubuntu 20.04 LTS server and a user with sudo permission (Not root).

I followed the steps in Readme and it works without comment out the `read: only` attribute.

I guess the folder permission might be the reason, I keeps user:group as the current log-in user except the path `/volumes/app/mattermost`.

**Comment by shurrman** [comment]
may be make it an env parameter?
